### ISSUE TITLE
New buttons to simplify use when creating levels

### ISSUE THREAD

**Issue by Dariosky-01** [TheJoeSmo/Foundry#207]
Salutations

I have a new proposal for the simplification of the use of Foundry during the creation of level to avoid uncomfortable behaviors, to be clearer, the hassle of going to look for similar objects with the obligation to always go and look.

PREMISE

To be able to make use of these commands there are
buttons which in practice automatically replace an object already present in the layer area, and which are available only if the object is single and is selected inside the layer.
Among them there is also a switch.

These buttons could be easily integrated into an internal window, and are compatible with the vision of the idea I had in publication # 205 where I proposed, for objects, a dynamic window that changed as needed according to the icon chosen, this window had to modeled according to the need, this modification allowed to obtain a significant reduction in the number of objects inside the left objects tab where the icons representing them are shown.
This window basically had to limit to a single icon the many duplicate icons inside the icon window that are currently used for dragging each single object of these which appear with the same appearance but which in reality, during the game started, serve different purposes with Mario's interaction with them.
This dynamic window would have the purpose, in the case of the "question block" objects, to let the user choose the function currently attributed to all the "question blocks" objects present in the multitudes of icons in the icon window in the objects tab by limiting there single-icon question blocks.
This obviously represents an advantage for the user because he does not have to search wasting time, it recognizes a single icon and if it has many features in the game this would be chosen by the user within this dynamic window.
I mean this window dynamic because it changes when the object in use occurs.
In the case of extensible blocks the window should bring up switches like the ones I talked about in publication # 205.

DESCRIPTION OF THE AMENDMENT OF THIS PUBLICATION.

The new buttons that I want to propose in this publication are commands that can very well be part of the dynamic window as buttons or switches that could always be present in a new window if desired, but they can be easily integrated into the dynamic window when the function is active for these buttons which I will list below.
Below I describe these new buttons and this single switch you will see described in its operation a bit as if I had to integrate it, thus describing how they should work, this way of describing them only serves to understand how they should integrate to work.
These buttons I want to point out that they can also work without the dynamic window I mentioned earlier that I would like there to be, so in this case they can be created very well in another area, in case, a new dedicated window that can be activated from the Foundry "view" menu .

DESCRIPTION OF THE BUTTONS AND NOTES OF OPERATING CONDITIONS BELOW

Key note:

"logical activation status"
=
If the element inside the layer is not a sprite or a "control object" then activate tools (here subordinate)

1) [Button]
Find the same mirror drawn object of the same size and replace it in the same position;

If a different object other than mirrored does not exist deactivates the button.

2) [Button]
Find the same object of the same size and rotated by + 90 ° and replace it in the same position, otherwise if there is no alter ego rotated by 90° rotate the object by 90° starting from its first block on the left (if for example a block of wood is placed horizontally) or from its first block at the top (if a wood block is placed vertically);

If the object is of size 1, block deactivates the button.

3) [Button]
Multiply by extending the selected block by double in the position where it is;

If the next multiplication leads to a number equal to or greater than the maximum extension number and, if the next step of this multiplication, during the active switch in "negative mode" leads to a number less than 1 then deactivate the button.

4) [Button]
Multiply by triple extending the selected block in the position where it is;

If the next multiplication leads to a number equal to or greater than the maximum extension number and, if the next step of this multiplication, during the active switch in "negative mode" leads to a number less than 1 then deactivate the button.

5) [Switch]
Activate "negative mode" of the multiplier.

If these features seem excessively very complicated for people new to Foundry these new buttons of this publication could meanwhile be integrated into the dynamic window mentioned in publication # 205 (with these appearing and disappearing depending on if they are active) and could be viewable in an "advanced" mode available as a choice in the Foundry options, where it therefore implies that all the changes to the editor that would be needed for the realization, including therefore the reduction of the multiple icons reduced to a single one, where the choice of the functionality of these would be delegated to the dynamic window and which would change the icon dragged into the layer according to and only if the dragged object is selected in the work area.

**Comment by TheJoeSmo** [comment]
I agree that grouping objects together would be preferred.  Then from a secondary menu you could select the specific version.  My concern is in which way does a series of objects qualify for a group.  And how ought one summon this menu?

**Comment by Dariosky-01** [comment]
Well it's simple, you have to be able to provide variables as if they were personal names to all internal features, separating the variables used to do what that functionality was intended or you have to do routines and sub routines called by the user to understand some settings and features instead of others.

Example: a function can be called Standard, Advanced as if they were two names like Dario, Walter (two names, even in real life it is like this among people) so if a function is included in the "advanced" mode it will be recalled and will be visible as is set.

Otherwise the names Stamdard and Advanced are routines called by the user that change, in this case all the data loaded into memory, most likely a software restart may be required because to act by decreasing the number of icons with this system you have to intervene at the base of the software data loaded into memory, replacing them because in this case we are talking about the data they are used to establish how many icons and how the icons should behave.
